beetsplug/acousticbrainz.py

```diff
@@ -41,64 +45,161 @@ def func(lib, opts, args):
         cmd.func = func
         return [cmd]
 
+    def import_task_files(self, session, task):
+        """Function is called upon beet import.
+        """
+
+        items = task.imported_items()
+        fetch_info(self._log, items)
+
 
 def fetch_info(log, items):
-    """Currently outputs MBID and corresponding request status code
+    """Currently outputs MBID and corresponding request status code.
     """
     for item in items:
         if item.mb_trackid:
             log.info('getting data for: {}', item)
 
             # Fetch the data from the AB API.
-            url = generate_url(item.mb_trackid)
-            log.debug('fetching URL: {}', url)
+            high_url = generate_url(item.mb_trackid, "/high-level")
+            low_url = generate_url(item.mb_trackid, "/low-level")
+            log.debug('fetching URLs: {} and {}', high_url, low_url)
             try:
-                rs = requests.get(url)
+                high = requests.get(high_url)
+                low = requests.get(low_url)
             except requests.RequestException as exc:
                 log.info('request error: {}', exc)
                 continue
 
             # Check for missing tracks.
-            if rs.status_code == 404:
+            if high.status_code == 404 or low.status_code == 404:
                 log.info('recording ID {} not found', item.mb_trackid)
                 continue
 
             # Parse the JSON response.
             try:
-                data = rs.json()
+                high_data = high.json()
+            except ValueError:
+                log.debug('Invalid Response from high-level: {}', high.text)
+            try:
+                low_data = low.json()
             except ValueError:
-                log.debug('Invalid Response: {}', rs.text)
+                log.debug('Invalid Response from low-level: {}', low.text)
 
             # Get each field and assign it on the item.
             item.danceable = get_value(
                 log,
-                data,
+                high_data,
                 ["highlevel", "danceability", "all", "danceable"],
             )
```

Comment:
If you made `get_value` an inner function, it would automatically have access to `log` and `high_data` (or `data` if you followed my other suggestion), so you wouldn't need to pass these in multiple times.